amm/contracts/stable_pool/lib.rs

```diff
@@ -566,6 +580,14 @@ pub mod stable_pool {
             amounts: Vec<u128>,
             to: AccountId,
         ) -> Result<(u128, u128), StablePoolError> {
+            //check amounts if it is initial liquidity supply
+            if self.total_supply() == 0 {
+                amounts
+                    .iter()
+                    .enumerate()
+                    .try_for_each(|(id, &amount)| self.check_min_reserve(id, amount))?;
```

Comment:
It looks like on pool's setup, we will store 12 as precision of stablecoins (18-6=12) and in `check_min_reserve` we will multiply `1*12 < 18` so we will need to provide at least 1M stablecoins on setup?

Reply:
The amounts are first adjusted to 18 decimal precision so with `MIN_RESERVE == 10^18` (which happens to be the common internal precision too) the minimum initial deposit would be exactly one token, e.g. 1 USDT and 1 USDC (`1_000_000` and `1_000_000`)

Reply:
Ah, I see now. I used 1 instead of 1_000_000 as "1 token".

BTW, how do we enforce that token decimals (passed in the [constructor](https://github.com/Cardinal-Cryptography/common-amm-stable-swap/blob/min-reserve/amm/contracts/stable_pool/lib.rs#L202)) are actually correct? Isn't it dangerous to let users define that? What if someone says USDC has 4 decimals? or 14?

Reply:
I'd say it is the creator's responsibility. If we want to enforce it, I see 2 possible ways, all of which require calling `PSP22Metadata::token_decimals()` on each token (assuming that the token contracts implement it):

1. Query/check token decimals in the constructor
2. Query/check token decimals in the Factory (when one exists)

Reply:
I don't think we have time to write the factory prior to the audit, so we're left with option 1. It's true that in theory tokens might not implement `PSP22Metadata` although all ours do. 

OK, let's push this responsibility to the creator and if he messes it up then it's his fault.